lib/mods/replace-get-set.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const nodeTypes = require('ts-morph').TypeGuards;
+const {updateImports} = require('../imports.js');
+const {
+  formatReplacement,
+  isGlobal,
+  getOriginal,
+} = require('../originals-helper.js');
+
+const dataProperties = new Map([
+  ['Array', ['length']],
+  ['String', ['length']],
+]);
+
+/**
+ * @param {string} type - the underlying type of the object
+ * @param {string} propertyName - the name of the property
+ *
+ * @return {boolean} whether the input is a data property and shouldn't be reset
+ */
+function isDataProperty(type, propertyName) {
+  return dataProperties.has(type) &&
+      dataProperties.get(type).includes(propertyName);
+}
+
+/**
+ * Replaces getset calls in the current node with the Get-Originals version
+ *
+ * @param {Node} node - a ts-morph AST Node
+ * @param {Object} imports - the current list of imports
+ */
+function replaceGlobalGetSetCalls(node, imports) {
+  if (nodeTypes.isPropertyAccessExpression(node) &&
+      !nodeTypes.isCallExpression(node.getParent()) &&
```

Comment:
This helps us see that it's not a method call, since they need to be treated differently from get-sets

```
// get/set
url.pathname;

// method call
url.toString();
```

Reply:
I'm still confused by this, the example provided is not against a global but I think maybe I understand now, you are trying to prevent `window.alert()` from becoming `Reflect_apply(Window_alert_get, window)()` ?

Is that correct? I don't see something in the get_set test that looks like the above, so maybe I've misunderstood.

Anyway, whatever the explanation is, it should be inline in the code. E.g.

```js
function replaceGlobalGetSetCalls(node, imports) {
  // Must be a property access on a global, e.g. window.location
  if (!(nodeTypes.isPropertyAccessExpression(node) &&
      isGlobal(node.getExpression()))) {
    return;
  }

  // Must not be a call e.g. window.alert().
  if (nodeTypes.isCallExpression(node.getParent())) {
    return;
  }
```